airo-camera-toolkit/airo_camera_toolkit/image_transforms/image_transform.py

```diff
@@ -0,0 +1,58 @@
+from abc import ABC
+from typing import Tuple, Union
+
+from airo_typing import NumpyFloatImageType, NumpyIntImageType, OpenCVIntImageType
+
+HWCImageType = Union[OpenCVIntImageType, NumpyFloatImageType, NumpyIntImageType]
```

Comment:
should we put these types in the airo-typing package? OR will they never be used in another package? Even then, I would prefer a separate types.py file, to avoid circular imports in the future?

Reply:
I'm leaving these choices up to you. I'm not super happy about the name of `HWCImageType`, the main point is just that `torch`-style image shapes are currently not supported.

Reply:
I am wondering if it would be better to just take numpy float images in all the transforms and leave conversions to the user? Because if you accept torch images, you should probably also accept cuda-tensors etc and this seems like too much complexity maybe? That would make the HWCType no longer needed?

Reply:
regarding the name, I think it is okay like this. `ChannelLastImageType` doesn't really cut it either... And this type is maybe indeed specific enough to keep it in the airo-camera-toolkit package

Reply:
> I am wondering if it would be better to just take numpy float images in all the transforms and leave conversions to the user? Because if you accept torch images, you should probably also accept cuda-tensors etc and this seems like too much complexity maybe? That would make the HWCType no longer needed?

@Victorlouisdg, what do you think about this?

Reply:
I agree that we shouldn't support torch-style images as this would increase the complexity of the implementation too much. 

However I don't really see a reason to only accept `NumpyFloatImageType`, rather than avoiding the `HWCImageType`. I don't believe it's worth the inconvenience (and inefficiency) of imposing a conversion on users. I do believe that we should add tests to ensure that all transforms treat `OpenCVIntImageType`, `NumpyFloatImageType`, `NumpyIntImageType` correctly.

Reply:
okay, if we test it I'm fine with it. It is indeed complexity of implementation vs efficiency. I would take reduced complexity but I can live with both :)